### PR TITLE
Cli: Fix create-with-seed

### LINKED ISSUE #8696: CLI: "create-address-with-seed" is broken
#### Problem
Command seems to return `SignerError::CustomError("Default keypair must be set if pubkey arg not provided".to_string())` regardless of inputs

#### Proposed Solution
Track down regression and fix

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
`solana create-with-seed ...` errors our no matter what from or config keypair is provided.
Also, users might want to derive keys from any kind of signer, including hardware wallets or seed phrases, and there is currently no support for that.

#### Summary of Changes
- Handle `--from` and config keypair setting appropriately to fix subcommand
- Add `pubkey_of_signer()` method to generate a pubkey from any kind of acceptable signer and use to extend create-with-seed functionality (we may want to use this method in other cli subcommands; will examine and pr separately)

Fixes #8696

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
@garious , after this lands, this works:
```
solana create-address-with-seed "test" STAKE --from usb://ledger
```

**Comment by garious** [comment]
Can't fix a regression without a test.

**Comment by CriesofCarrots** [comment]
> Can't fix a regression without a test.

✔️ 
(Not sure why github is showing my commits out of order)

File: cli/src/cli.rs [hunk 1]
```diff
@@ -1070,9 +1084,12 @@ fn process_create_address_with_seed(
     seed: &str,
     program_id: &Pubkey,
 ) -> ProcessResult {
-    let config_pubkey = config.pubkey()?;
-    let from_pubkey = from_pubkey.unwrap_or(&config_pubkey);
-    let address = create_address_with_seed(from_pubkey, seed, program_id)?;
+    let from_pubkey = if let Some(pubkey) = from_pubkey {
```
Comment (t-nelson) [inline cli/src/cli.rs]: NIT: `unwrap_or()`

Comment (CriesofCarrots) [reply cli/src/cli.rs]: Can't use `?` in a closure that returns a Pubkey, so the current construction seems simpler to me for the `else` case.

**Pull request merged**